### PR TITLE
Dor::Identifiable#get_related_obj_display_title fix, empty datastream tests

### PR DESCRIPTION
**Pull request by jmartin-sul**
* identifiable.rb:  get_related_obj_display_title should use descMetadata title. 
 * identifiable_spec.rb:  update tests accordingly.
* desc_metadata_spec.rb:  add tests for empty datastream scenarios.  important to this fix because get_related_obj_display_title now relies on that behavior.

closes issue #196

### CODE REVIEW

File: lib/dor/models/identifiable.rb [hunk 1]
```diff
@@ -209,15 +209,13 @@ def update_tag(old_tag, new_tag)
...
-          return related_obj.label
-        end
+      # desc_md_ds_title will be nil if related_obj is nil
+      desc_md_ds_title = if related_obj
+        desc_md_ds = related_obj.datastreams['descMetadata']
+        desc_md_ds ? desc_md_ds.title_info.main_title[0] : nil
       end
 
-      default_title
+      !desc_md_ds_title.blank? ? desc_md_ds_title : default_title
```
Comment (drh-stanford) [inline lib/dor/models/identifiable.rb]: Use `.present?` instead of `! .blank?` ?

File: lib/dor/models/identifiable.rb [hunk 2]
```diff
@@ -209,15 +209,13 @@ def update_tag(old_tag, new_tag)
...
-      if related_obj
-        if related_obj.datastreams['DC'] && related_obj.datastreams['DC'].title
-          return related_obj.datastreams['DC'].title
-        else
-          return related_obj.label
-        end
+      # desc_md_ds_title will be nil if related_obj is nil
+      desc_md_ds_title = if related_obj
+        desc_md_ds = related_obj.datastreams['descMetadata']
+        desc_md_ds ? desc_md_ds.title_info.main_title[0] : nil
```
Comment (drh-stanford) [inline lib/dor/models/identifiable.rb]: I'd use `.first` rather than `[0]` for readability.

**Comment by drh-stanford** [comment]
Other than a couple nits I'm :+1: for this.

File: lib/dor/models/identifiable.rb (hunk 1 above)
Comment (jmartin-sul) [inline lib/dor/models/identifiable.rb]: cool, didn't know about `present?`.

**Comment by jmartin-sul** [comment]
nits picked
